evm/src/cpu/kernel/interpreter.rs

```diff
@@ -120,17 +123,46 @@ pub(crate) fn run_interpreter_with_memory(
     Ok(interpreter)
 }
 
-pub(crate) fn run<'a>(
+pub(crate) fn run<'a, F: Field>(
     code: &'a [u8],
     initial_offset: usize,
     initial_stack: Vec<U256>,
     prover_inputs: &'a HashMap<usize, ProverInputFn>,
-) -> anyhow::Result<Interpreter<'a>> {
+) -> anyhow::Result<Interpreter<'a, F>> {
     let mut interpreter = Interpreter::new(code, initial_offset, initial_stack, prover_inputs);
     interpreter.run()?;
     Ok(interpreter)
 }
 
+/// Simulates the CPU execution from `state` until the program counter reaches `final_label`  
+/// in the current context.
+pub(crate) fn simulate_cpu_and_get_user_jumps<F: Field>(
+    final_label: &str,
+    state: &GenerationState<F>,
+) -> Option<HashMap<usize, Vec<usize>>> {
+    if state.jumpdest_table.is_some() {
+        None
+    } else {
+        let halt_pc = KERNEL.global_labels[final_label];
+        let initial_context = state.registers.context;
+        let mut interpreter =
+            Interpreter::new_with_state_and_halt_condition(state, halt_pc, initial_context);
+
+        log::debug!("Simulating CPU for jumpdest analysis.");
+
+        interpreter.run().unwrap();
```

Comment:
I think this method should change to return `Result<Option<HashMap<usize, Vec<usize>>>, SomeError>`. Not ok to panic here I think.

Reply:
Yeah agree, let's gently error during simulations, although I think there may be other unwraps on the interpreter side

Reply:
Yes, I was only doing this for debugging purposes

Reply:
Actually, here I need to always return `interpreter.jumpdest_table`, even when the simulation fails.  I guess the best option is to ignore any error in `run()`